docs/tutorials/9_compare_baselines.ipynb

```diff
@@ -0,0 +1,429 @@
+   "metadata": {},
+   "source": [
+    "But wait! We only ran the evaluation once. What if we got lucky? Let's run the evaluation a few more times and see what happens."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "expert_reward, _ = evaluate_policy(expert, env, 10)\n",
+    "not_expert_reward, _ = evaluate_policy(not_expert, env, 10)\n",
+    "\n",
+    "print(f\"Expert reward: {expert_reward:.2f}\")\n",
+    "print(f\"Not expert reward: {not_expert_reward:.2f}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Seems a bit more robust now, but how certain are we? Fortunately, `imitation` provides us with tools to answer this.\n",
+    "\n",
+    "We will perform a permutation test using the `is_significant_reward_improvement` function. We want to be very certain -- let's set the bar high and require a p-value of 0.001."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from imitation.testing.reward_improvement import is_significant_reward_improvement\n",
+    "\n",
+    "expert_rewards, _ = evaluate_policy(expert, env, 10, return_episode_rewards=True)\n",
+    "not_expert_rewards, _ = evaluate_policy(\n",
+    "    not_expert, env, 10, return_episode_rewards=True\n",
+    ")\n",
+    "\n",
+    "significant = is_significant_reward_improvement(\n",
+    "    not_expert_rewards, expert_rewards, 0.001\n",
+    ")\n",
+    "\n",
+    "print(\n",
+    "    f\"The expert is {'NOT ' if not significant else ''}significantly better than the not-expert.\"\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Huh, turns out we set the bar too high. We could lower our standards, but that's for cowards.\n",
+    "Instead, we can collect more data and try again."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from imitation.testing.reward_improvement import is_significant_reward_improvement\n",
+    "\n",
+    "expert_rewards, _ = evaluate_policy(expert, env, 100, return_episode_rewards=True)\n",
+    "not_expert_rewards, _ = evaluate_policy(\n",
+    "    not_expert, env, 100, return_episode_rewards=True\n",
+    ")\n",
+    "\n",
+    "significant = is_significant_reward_improvement(\n",
+    "    not_expert_rewards, expert_rewards, 0.001\n",
+    ")\n",
+    "\n",
+    "print(f\"Is the expert significantly better than the not-expert? {significant}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we go! We can now be 99.9% confident that the expert is better than the not-expert. It might still be an extraordinary stroke of luck, or a conspiracy to make us choose the wrong algorithm, but outside of that, we can be pretty sure that the expert is better."
```

Comment:
This is true in terms of these two specific policies. However, it might be worth noting that this does *not* show that training with 100,000 timesteps is better than 10,000 timesteps with 99.9% confidence -- we only have a single sample of the *training run*, even if we have many samples of the *rollouts* from those policies (holding the training run fixed).

I'm sure you understand this, but it's a misinterpretation I think someone could easily make, so it may be worth explicitly flagging.

Reply:
I added a bunch of commentary across the notebook to address this, hopefully it's enough